autoendpoint/src/server.rs

```diff
@@ -47,11 +51,25 @@ impl Server {
         let bind_address = format!("{}:{}", settings.host, settings.port);
         let fernet = settings.make_fernet();
         let endpoint_url = settings.endpoint_url();
-        let ddb = Box::new(DbClientImpl::new(
-            metrics.clone(),
-            settings.router_table_name.clone(),
-            settings.message_table_name.clone(),
-        )?);
+        let db_settings = DbSettings {
+            dsn: settings.db_dsn.clone(),
+            db_settings: if settings.db_settings.is_empty() {
+                warn!("❗ Using obsolete message_table and router_table args");
+                // backfill from the older arguments.
+                json!({"message_table": settings.message_table_name, "router_table":settings.router_table_name}).to_string()
+            } else {
+                settings.db_settings.clone()
+            },
+        };
+        let ddb: Box<dyn DbClient> = match StorageType::from_dsn(&db_settings.dsn) {
+            StorageType::DynamoDb => Box::new(
+                DdbClientImpl::new(metrics.clone(), &db_settings)
+                    .map_err(ApiErrorKind::Database)?,
+            ),
+            StorageType::INVALID => {
```

Comment:
This will break the deployed autoendpoint because it doesn't set a `db_dsn`. I'm not sure how this will work as it is currently as setting `db_dsn` currently seems to imply the use of the emulator, as it causes main to set `AWS_LOCAL_DYNAMODB`.

I think we need some way of signaling: use dynamodb/no emulator stuff/use the autolookup from `rusoto_core::Region::default()`

Reply:
I'll make `StorageType::from_dsn` highly preferential to DynamoDb for now, and add a comment.